tfhe/src/integer/server_key/radix_parallel/vector_comparisons.rs

```diff
@@ -0,0 +1,108 @@
+use crate::integer::{BooleanBlock, IntegerRadixCiphertext, ServerKey};
+
+use rayon::prelude::*;
+
+impl ServerKey {
+    /// Compares two slices containing ciphertexts and returns an encryption of `true` if all
+    /// pairs are equal, otherwise, returns an encryption of `false`.
+    ///
+    /// - Slices must have the same length
+    /// - Pairs (`lhs[i]`, `rhs[i]`) must have the same number of blocks
+    pub fn unchecked_all_eq_slices_parallelized<T>(&self, lhs: &[T], rhs: &[T]) -> BooleanBlock
+    where
+        T: IntegerRadixCiphertext,
+    {
+        assert_eq!(
+            lhs.len(),
+            rhs.len(),
+            "lhs and rhs slices must have same len (lhs: {}, rhs: {})",
+            lhs.len(),
+            rhs.len()
+        );
+        if lhs
+            .iter()
+            .zip(rhs.iter())
+            .any(|(l, r)| l.blocks().len() != r.blocks().len())
+        {
+            panic!("All pairs of element in the lhs and rhs slices must have the same number of blocks");
+        }
```

Comment:
the different length is a trivial case of False IMO

the other loop it could also be a trivial case of false potentially given some pairs are not of the same type/under the same modulus, WDYT ?

Reply:
Yes non matching len for slices should be  trvial false I agree

For the mismatched len between pairs, I'm not so sure, it depends on how strict we want to be, because technically, two unsigend radix with different could encrypt the same value.

But returning false might be better than panic, so we could return false and document that

Reply:
I think I would document, worst case the HL would not suffer from it as it's going to use "uniform" arrays right ?

so to have something peculiar at the integer level might be kind of ok, not sure if it follows the rest of the design on mismatched sizes though